dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -83,6 +84,58 @@ public abstract class ClientBase
         return response.Value.Choices[0].Text;
     }
 
+    protected async IAsyncEnumerable<string> InternalCompletionStreamAsync(
+        string text,
+        CompleteRequestSettings requestSettings,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(requestSettings);
+
+        if (requestSettings.MaxTokens < 1)
+        {
+            throw new AIException(
+                AIException.ErrorCodes.InvalidRequest,
+                $"MaxTokens {requestSettings.MaxTokens} is not valid, the value must be greater than zero");
+        }
+
+        var options = new CompletionsOptions
+        {
+            Prompts = { text.NormalizeLineEndings() },
+            MaxTokens = requestSettings.MaxTokens,
+            Temperature = (float?)requestSettings.Temperature,
+            NucleusSamplingFactor = (float?)requestSettings.TopP,
+            FrequencyPenalty = (float?)requestSettings.FrequencyPenalty,
+            PresencePenalty = (float?)requestSettings.PresencePenalty,
+            Echo = false,
+            ChoicesPerPrompt = 1,
+            GenerationSampleCount = 1,
+            LogProbabilityCount = null,
+            User = null,
+        };
+
+        if (requestSettings.StopSequences is { Count: > 0 })
+        {
+            foreach (var s in requestSettings.StopSequences)
+            {
+                options.StopSequences.Add(s);
+            }
+        }
```

Comment:
I assume this is the same options setup across both CompleteAsync and CompletionStreamAsync (line 101 to 122). If so, can you make that a private helper to that future fixes are picked up in both calls?

Reply:
Good point, moved repeated code to private methods.